modules/tinymce/src/plugins/paste/test/ts/atomic/FragmentParserTest.ts

```diff
@@ -1,149 +1,89 @@
+      assert.deepEqual(simple, { html: '<LI>abc</LI>', context: 'ul' }, 'simple list');
+
+      const newLines = FragmentParser.getFragmentInfo('<BODY>\n<UL>\n<!--StartFragment-->\n<LI>abc</LI>\n<!--EndFragment-->\n</UL>\n</BODY>');
+      assert.deepEqual(newLines, { html: '\n<LI>abc</LI>\n', context: 'ul' }, 'list with new lines');
+    });
+
+    it('should be a p context and the content inside the fragment markers for paragraphs', () => {
+      const result = FragmentParser.getFragmentInfo('<BODY><P><!--StartFragment--><B>abc</B><!--EndFragment--></P></BODY>');
+      assert.deepEqual(result, { html: '<B>abc</B>', context: 'p' });
+    });
+
+    it('should be a h<num> context and the content inside the fragment markers for headings', () => {
+      const result = FragmentParser.getFragmentInfo('<BODY><H1><!--StartFragment--><B>abc</B><!--EndFragment--></H1></BODY>');
+      assert.deepEqual(result, { html: '<B>abc</B>', context: 'h1' });
+    });
+  });
+
+  context('getFragmentHtml', () => {
+    it('should be the input content', () => {
+      const result = FragmentParser.getFragmentHtml('abc');
+      assert.equal(result, 'abc');
+    });
+
+    it('should be the content inside the fragment markers for text', () => {
+      const text = FragmentParser.getFragmentHtml('<!-- StartFragment -->abc<!-- EndFragment -->');
+      assert.equal(text, 'abc', 'text with no before/after content');
+
+      const textNoSpaces = FragmentParser.getFragmentHtml('<!--StartFragment-->abc<!--EndFragment-->');
+      assert.equal(textNoSpaces, 'abc', 'text with no before/after content and no spaces in comments');
+
+      const textWithOuterContent = FragmentParser.getFragmentHtml('X<!--StartFragment-->abc<!--EndFragment-->Y');
+      assert.equal(textWithOuterContent, 'abc', 'text with before/after content');
+    });
+
+    it('should be the content inside the fragment markers for html', () => {
+      const result = FragmentParser.getFragmentHtml('<!DOCTYPE html><BODY><!-- StartFragment --><B>bold</B><I><B>abc</B>This</I><!-- EndFragment --></BODY></HTML>');
+      assert.equal(result, '<B>bold</B><I><B>abc</B>This</I>');
+    });
+
+    it('should be the content inside the fragment markers for lists', () => {
+      const simple = FragmentParser.getFragmentHtml('<BODY><UL><!--StartFragment--><LI>abc</LI><!--EndFragment--></UL></BODY>');
+      assert.deepEqual(simple, '<LI>abc</LI>', 'simple list');
+
+      const newLines = FragmentParser.getFragmentHtml('<BODY>\n<UL>\n<!--StartFragment-->\n<LI>abc</LI>\n<!--EndFragment-->\n</UL>\n</BODY>');
+      assert.deepEqual(newLines, '\n<LI>abc</LI>\n', 'list with new lines');
+
+      const withRoot = FragmentParser.getFragmentHtml('<BODY CLASS="x"><!--StartFragment--><UL><LI>abc</LI></UL><!--EndFragment--></BODY>');
+      assert.deepEqual(withRoot, '<UL><LI>abc</LI></UL>', 'list with root element');
```

Comment:
Do these three need to be `deepEqual`?

Reply:
They do not... I just copied the above and forgot to change them. Thanks!

Reply:
Fixed